src/provider/LocalStorage/ConfigurationServiceLocalStorage.ts

```diff
@@ -106,20 +107,51 @@ export class ConfigurationServiceLocalStorage<T=any> implements ConfigurationSer
     if (repositoryRequestValidator(request)) {
       return Promise.reject(new Error(messages.repositoryNotProvided));
     }
-  
+
     if (repositoryKeyRequestValidator(request)) {
       return Promise.reject(new Error(messages.repositoryKeyNotProvided));
     }
-    
+
     return new Promise((resolve, reject) => {
       this.getRepository(request.repository).then((repository) => {
-        localStorage.setItem(`${this.token}.${request.repository}`, JSON.stringify({
-          ...repository,
-          [request.key]: request.value
-        }));
-        
+        this.updateRepositoryData(request, repository);
         resolve({});
       }).catch(reject);
     });
   }
+
+  public createEntry(request: SaveRequest<T>) {
+    return this.changeEntry(request, 'create');
+  };
+
+  public updateEntry(request: SaveRequest<T>) {
+    return this.changeEntry(request, 'update');
+  };
+
+  private changeEntry(request: SaveRequest<T>, mode: 'create' | 'update') {
+    try {
+      validateSaveEntryRequest(request);
```

Comment:
```suggestion
      validateChangeEntryRequest(request);
```

Reply:
the type name is SaveRequest - the method name was based on this.